// @argos/utils/esnext/utils.js
import agnostify from './agnostify'
import { animate } from './animate'
import bem from './bem'
import canUseDom from './canUseDom'
import debounce from './debounce'
import formatPartNumber from './formatPartNumber'
import isEmpty from './isEmpty'
import randomString from './randomString'
import removeKnownProps from './removeKnownProps'
import smooth from './smooth-scroll'
import getUserAgent from './getUserAgent'
import {
  getSearchPath,
  getBrowsePath,
  getCategoryFilterPath,
  getOptionsPath,
  getRefinementPath,
  getEncodedBrowseCategories,
  getCategoriesJoined,
  refinementSort,
  refinementKeySort,
  hyphenateSpaces,
  stripPunctuation,
  encode,
  decode,
  getSanitised,
} from './findabilityUrlHelper'

import { defaultMapper, sortAttributes, sortVariants, typeMapper, variantFilter } from './sortVariants'

export {
  agnostify,
  animate,
  bem,
  canUseDom,
  debounce,
  defaultMapper,
  getUserAgent,
  formatPartNumber,
  isEmpty,
  randomString,
  removeKnownProps,
  smooth,
  sortAttributes,
  sortVariants,
  typeMapper,
  variantFilter,
  getSearchPath,
  getBrowsePath,
  getCategoryFilterPath,
  getOptionsPath,
  getRefinementPath,
  getEncodedBrowseCategories,
  getCategoriesJoined,
  refinementSort,
  refinementKeySort,
  hyphenateSpaces,
  stripPunctuation,
  encode,
  decode,
  getSanitised,
}
